python/lsst/daf/butler/core/formatter.py

```diff
@@ -141,6 +141,31 @@ class FormatterFactory:
     def __init__(self):
         self._mappingFactory = MappingFactory(Formatter)
 
+    def normalizeDimensions(self, universe):
+        """Normalize formatter lookups that use dimensions.
+
+        Parameters
+        ----------
+        universe : `DimensionGraph`
+            The set of all known dimensions. If `None`, returns without
+            action.
+
+        Notes
+        -----
+        Goes through all registered formatters, and for keys that include
+        dimensions, rewrites those keys to use a verified set of
+        dimensions.
+
+        Returns without action if the formatter keys have already been
+        normalized.
+
+        Raises
+        ------
+        ValueError
+            A key exists where a dimension is not part of the ``universe``.
```

Comment:
I thought we were supposed to start these sentences with something like "Raised if ...", but I could be mistaken.

Reply:
You are right. I never noticed that rules before.